### PR TITLE
fix(email-inbound-connector): implement a reconnect and reopen mechanism

### PR DESCRIPTION
**Pull request by mathias-vandaele**
## Description

As asked by @sbuettner this PR implements a re connect / re open mechanism

## Related issues

closes #

## Checklist

- [x] PR has a **milestone** or the `no milestone` label.

### CODE REVIEW

File: connectors/email/src/main/java/io/camunda/connector/email/client/jakarta/inbound/PollingManager.java [hunk 1]
```diff
@@ -78,12 +82,30 @@ public static PollingManager create(
...
       case PollUnseen pollUnseen -> pollUnseenAndProcess(pollUnseen);
     }
   }
 
+  private void prepareForPolling() {
+    if (!this.store.isConnected()) {
+      try {
+        this.jakartaUtils.connectStore(store, authentication);
+      } catch (MessagingException e) {
+        throw new RuntimeException("Could not reconnect to store");
```
Comment (sbuettner) [inline connectors/email/src/main/java/io/camunda/connector/email/client/jakarta/inbound/PollingManager.java:97]: Can you add the exception to the log so we can find out what failed?

File: connectors/email/src/main/java/io/camunda/connector/email/client/jakarta/inbound/PollingManager.java [hunk 2]
```diff
@@ -78,12 +82,30 @@ public static PollingManager create(
...
+      }
+    }
+    if (!this.folder.isOpen()) {
+      try {
+        this.folder.open(Folder.READ_WRITE);
+      } catch (MessagingException e) {
+        throw new RuntimeException("Could not reopen folder");
```
Comment (sbuettner) [inline connectors/email/src/main/java/io/camunda/connector/email/client/jakarta/inbound/PollingManager.java:104]: same as above

File: connectors/email/src/main/java/io/camunda/connector/email/client/jakarta/outbound/JakartaEmailActionExecutor.java [hunk 3]
```diff
@@ -130,7 +130,7 @@ private MoveEmailResponse imapMoveEmails(
       sourceImapFolder.open(Folder.READ_WRITE);
       Folder targetImapFolder =
           store.getFolder(
-              String.join(String.valueOf(rootFolder.getSeparator()), toFolder.split("\\.")));
+              String.join(String.valueOf(rootFolder.getSeparator()), toFolder.split("[./]")));
```
Comment (sbuettner) [inline connectors/email/src/main/java/io/camunda/connector/email/client/jakarta/outbound/JakartaEmailActionExecutor.java:133]: Can we use a constant for this.

File: connectors/email/src/main/java/io/camunda/connector/email/client/jakarta/utils/JakartaUtils.java [hunk 4]
```diff
@@ -321,7 +321,7 @@ public void moveMessage(Store store, Message message, String targetFolder) {
       char separator = imapFolder.getSeparator();
       String targetFolderFormatted =
           Optional.ofNullable(targetFolder)
-              .map(string -> string.split("\\."))
+              .map(string -> string.split("[./]"))
```
Comment (sbuettner) [inline connectors/email/src/main/java/io/camunda/connector/email/client/jakarta/utils/JakartaUtils.java:324]: ... and use this constant here as well?

Or create a simple static method for this logic. Would be even better

File: connectors/email/src/main/java/io/camunda/connector/email/client/jakarta/utils/JakartaUtils.java [hunk 5]
```diff
@@ -321,9 +310,9 @@ public void moveMessage(Store store, Message message, String targetFolder) {
       char separator = imapFolder.getSeparator();
       String targetFolderFormatted =
           Optional.ofNullable(targetFolder)
-              .map(string -> string.split("\\."))
+              .map(string -> string.split(REGEX_PATH_SPLITTER))
               .map(strings -> String.join(String.valueOf(separator), strings))
-              .orElse("temp");
+              .orElseThrow(() -> new RuntimeException("No folder has been set"));
```
Comment (johnBgood) [inline connectors/email/src/main/java/io/camunda/connector/email/client/jakarta/utils/JakartaUtils.java:315]:
```suggestion
              .orElseThrow(() -> new RuntimeException("No target folder has been set"));
```

**Review by johnBgood — APPROVED** [review]
🔥

**Comment by sbuettner** [comment]
Backport failed for `release/8.6`, because it was unable to cherry-pick the commit(s).

Please cherry-pick the changes locally and resolve any conflicts.
```bash
git fetch origin release/8.6
git worktree add -d .worktree/backport-3506-to-release/8.6 origin/release/8.6
cd .worktree/backport-3506-to-release/8.6
git switch --create backport-3506-to-release/8.6
git cherry-pick -x be06a0ee548f5049437bf24b54b2e730cdf1c582
```

**Pull request merged**